packages/story-editor/src/components/inspector/inspectorProvider.js

```diff
@@ -36,11 +36,12 @@ import { useFeature } from 'flagged';
 import { useAPI } from '../../app/api';
 import { useStory } from '../../app/story';
 import { useHighlights } from '../../app/highlights';
-import { DOCUMENT, DESIGN, PUBLISH_MODAL_DOCUMENT } from './constants';
+import { DOCUMENT, STYLE, PUBLISH_MODAL_DOCUMENT, INSERT } from './constants';
 import Context from './context';
 import DesignInspector from './design';
+import Library from '../library';
 
-const INSPECTOR_TAB_IDS = new Set([DOCUMENT, DESIGN]);
+const INSPECTOR_TAB_IDS = new Set([INSERT, DOCUMENT, STYLE]);
```

Comment:
Should this whole thing be behind a feature flag for now? I feel like a lot of stuff is being changed without being thoroughly tested and I also feel like this is only the first of many PR's implementing this change (follow-up refactors and design rework incoming), so perhaps it should be hidden? I know it's a lot of extra work, but maybe not too much extra work?

Well, perhaps except for the highlight stuff, that might be a bit tricky.

All the karma tests can be updated to use the feature flag, so they don't have to be reverted again.

Reply:
the highlighting update could probably be handled conditionally, if `subTab` or whatever it gets called is present then use that, if not fallback to the existing structure. I think that would make feature flagging doable? 

It might also just be something that has to look a little weird for like a week, if that's a thing everyone is ok with - pea pod can help out too to make it move faster. I agree with @barklund that this whole thing needs to be tested thoroughly.  Might be worth a slack thread to figure out a plan for the whole update since it's such a lift.

Reply:
Just to keep conversations current, discussion on feature flag was moved to [Slack](https://xwp.slack.com/archives/C0135V8LYJE/p1646317678228379?thread_ts=1646235979.968759&cid=C0135V8LYJE) and decided against because it adds unnecessary time to this effort. The plan is to keep this branch current with main and merge this and the floating menus together.